Adds diagrams to the multi-key how it works page (#1064)
BitcoinDesign/Guide: guide/how-it-works/private-key-management/multi-key.md

All of the previous schemes have relied on a single private key to control the wallet. This presents an all-or-nothing risk for loss of funds from both theft and negligence. To counter this, a wallet can have several private keys attached of which all or a subset need to sign any transactions.

> This is often called multi-signature, or multi-sig for short, but is also sometimes referred to as a *vault*. A multi-key setup is described as *n-of-n* to indicate how many keys are needed to sign a transaction out of the issued number. For example, a *2-of-3* setup requires two of the three private keys to sign a transaction for it to be valid.

Reviewer: ```suggestion
This is often called multi-signature, or multi-sig for short, but is also sometimes referred to as a *vault*. A multi-key setup is described as *m-of-n* to indicate how many keys are needed to sign a transaction out of the issued number. For example, a *2-of-3* setup requires two of the three private keys to sign a transaction for it to be valid.
```

{% include fact/close.html %}

### General user flow

Alice, Robert, and Charles want to set up a multi-key wallet that has a total of 3 keys, with 2 keys required to send a transaction.

> In the first step, each of them independently generate a private key and a public key (technically an extended public key). They will use the private key for initiating and approving transactions, and will never share it it. The public key is used for construct the multi-key wallet for generating addresses to receive bitcoin to.

Reviewer: Don't think the private key is required for "initiating" a transaction, only for signing/approving. Perhaps I misunderstand what you mean here...

Author: I don't see where it says that in the text...?

> In the first step, each of them independently generate a private key and a public key (technically an extended public key). They will use the private key for initiating and approving transactions, and will never share it it. The public key is used for construct the multi-key wallet for generating addresses to receive bitcoin to.

Reviewer: ```suggestion
In the first step, each of them independently generate a private key and a public key (technically an extended public key). They will use the private key for approving transactions, and will never share it it. The public key is shared with the other signers, and all the public keys are together used to generate addresses to receive bitcoin to.
```

> In the first step, each of them independently generate a private key and a public key (technically an extended public key). They will use the private key for initiating and approving transactions, and will never share it it. The public key is used for construct the multi-key wallet for generating addresses to receive bitcoin to.

Reviewer 2: ```suggestion
In the first step, each of them independently generate a private key and a public key (technically an extended public key). They will use the private key for initiating and approving transactions, and will never share it it. The public key is used to construct the multi-key wallet and for generating addresses to receive bitcoin to.
```

> In the first step, each of them independently generates a private key and a public key (technically an extended public key). All 3 public keys together are used to construct the multi-key wallet, for generating addresses to receive bitcoin to, and in extension also for constructing new transactions. The private key is never shared, and used to approve transactions, initiated by themselves or others.

Reviewer: nit

```suggestion
In the first step, each of them independently generates a private key and a public key (technically an extended public key). All 3 public keys together are used to construct the multi-key wallet, for generating addresses to receive bitcoin to, and by extension also for constructing new transactions. The private key is never shared, and used to approve transactions, initiated by themselves or others.
```

   mobile = "/assets/images/guide/how-it-works/private-key-management/multi-key/multi-key-diagram-1-key-generation-mobile.png"
   mobileRetina = "/assets/images/guide/how-it-works/private-key-management/multi-key/multi-key-diagram-1-key-generation-mobile@2x.png"
   alt-text = "Diagram of Alice, Robert, and Charles generating private and public keys"
   width = 800
   height = 456
%}

Then, they exchange the public keys, so each of them has a set of all 3.

{% include picture.html
   image = "/assets/images/guide/how-it-works/private-key-management/multi-key/multi-key-diagram-1-key-exchange.png"
   retina = "/assets/images/guide/how-it-works/private-key-management/multi-key/multi-key-diagram-1-key-exchange@2x.png"
   mobile = "/assets/images/guide/how-it-works/private-key-management/multi-key/multi-key-diagram-1-key-exchange-mobile.png"
   mobileRetina = "/assets/images/guide/how-it-works/private-key-management/multi-key/multi-key-diagram-1-key-exchange-mobile@2x.png"
   alt-text = "Diagram of the three wallet owners exchanging public keys"
   width = 800
   height = 381
%}

> They can now individually construct the same multi-key wallet with the 3 public keys and the spending conditions they decided on (2-of-3). Addresses generated by each wallet will be identical. The order of the public keys is important here. A different order will result in a different wallet.

Reviewer: The order of the public keys can be ensured by the software, so it is not necessary for the signers themselves to sign in any particular order. Unsure if this necessitates any change to the text here.

Reviewer 2: I think this would be a good addition.

Author: The order has to be the same for constructing the wallet, but not for signing. I tested this with the [BitcoinJS library](https://github.com/bitcoinjs/bitcoinjs-lib) earlier this week.

In the wallet coordination file, there are fingerprints of each key. That's how the order can be tracked later once the wallet is set up.

Author: I added an extra note about this.

- Make sure the multi-key setup itself is backed up properly, including [extended public keys]({{ '/guide/glossary/#extended-public-key-xpub-ypub-zpub' | relative_url }}) for all the participating keys, fingerprint and derivation.

**Products that use this scheme**
- [Sparrow Wallet](https://sparrowwallet.com/)
- [Casa](https://keys.casa) co-managed 2-of-3, or 3-of-5
- [Electrum](https://electrum.org)
- [BlueWallet](https://bluewallet.io)
- [Unchained Capital/Caravan](https://unchained-capital.com) co-managed
- [Specter](https://specter.solutions)
- [Armory](https://btcarmory.com)
- [Revault](https://revault.dev){:target="_blank"} - in development

> 

Reviewer 2: Maybe Nunchuk would also be a nice addition?

Author: Added.
